assets/js/stripe.js

```diff
@@ -818,48 +812,65 @@ jQuery( function( $ ) {
 		 * Those redirects/hashes are generated in `WC_Gateway_Stripe::process_payment`.
 		 */
 		onHashChange: function() {
-			var partials = window.location.hash.match( /^#?confirm-pi-(.+):(.+)$/ );
+			var partials = window.location.hash.match( /^#?confirm-pi-([^:]+):(.+)$/ );
 
 			if ( ! partials || 3 > partials.length ) {
 				return;
 			}
 
 			var intentClientSecret = partials[1];
-			var successRedirectURL = decodeURIComponent( partials[2] );
+			var redirectURL        = decodeURIComponent( partials[2] );
 
 			// Cleanup the URL
 			window.location.hash = '';
 
-			wc_stripe_form.openIntentModal( intentClientSecret, successRedirectURL );
+			wc_stripe_form.openIntentModal( intentClientSecret, redirectURL );
+		},
+
+		maybeConfirmIntent: function() {
+			if ( ! $( '#stripe-intent-id' ).length || ! $( '#stripe-intent-return' ).length ) {
+				return;
+			}
+
+			var intentSecret = $( '#stripe-intent-id' ).val();
+			var returnURL    = $( '#stripe-intent-return' ).val();
+
+			wc_stripe_form.openIntentModal( intentSecret, returnURL, true );
 		},
 
 		/**
-		 * Opens the modal for PaymentIntents.
+		 * Opens the modal for PaymentIntent authorizations.
 		 *
-		 * @param {string} intentClientSecret The client secret of the intent.
-		 * @param {string} successRedirectURL A URL to redirect to on success.
-		 * @param {string} failureRedirectURL A URL to redirect to on error (optional).
+		 * @param {string}  intentClientSecret The client secret of the intent.
+		 * @param {string}  redirectURL        The URL to ping on fail or redirect to on success.
+		 * @param {boolean} alwaysRedirect     If set to true, an immediate redirect will happen no matter the result.
+		 *                                     If not, an error will be displayed on failure.
 		 */
-		openIntentModal: function( intentClientSecret, successRedirectURL, failureRedirectURL ) {
+		openIntentModal: function( intentClientSecret, redirectURL, alwaysRedirect ) {
 			stripe.handleCardPayment( intentClientSecret )
 				.then( function( response ) {
 					if ( response.error ) {
 						throw response.error;
 					}
 
-					if ( 'succeeded' === response.paymentIntent.status ) {
-						window.location = successRedirectURL;
+					if ( 'succeeded' !== response.paymentIntent.status ) {
+						return;
 					}
+
+					window.location = redirectURL;
```

Comment:
Have you encountered any problem when doing a `window.location` redirect to an AJAX endpoint?  Would it still work if the `&is_ajax` suffix was always added? I'm thinking that redirecting to an AJAX endpoint is a hack and may cause some problems, but if it doesn't, I'm fine with it.